src/webapp/common.py

```diff
@@ -253,3 +253,15 @@ def load_yaml_file(filename) -> Dict:
     except yaml.YAMLError as e:
         log.error("YAML error in %s: %s", filename, e)
         raise
+
+
+def readfile(path, logger):
+    """ return stripped file contents, or None on errors """
+    if path:
+        try:
+            with open(path, mode="rb") as f:
+                return f.read().strip()
+        except IOError as e:
+            logger.error("Failed to read file '%s': %s" % (path, e))
```

Comment:
```suggestion
            logger.error("Failed to read file '%s': %s", path, e)
```
at least if it's a standard logging object. This is the general pattern used elsewhere; the advantage is that if no output will be printed because of the log level, then Python doesn't bother with the substitution.

Doesn't _matter_ too much in this case, except for consistency's sake.

Reply:
Also you should do `if logger:` first, in case anybody wants to call readfile without a logger.